### PR TITLE
Improve logging (Loguru)

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
- [x] Test with real sensor data inside raspberry pi

example:
image

### CODE REVIEW

File: services/data_logger/main.py [hunk 1]
```diff
@@ -194,19 +183,62 @@ def main(args: argparse.Namespace):
...
+
+
+def configure_logging(args: argparse.Namespace):
+    loguru_output_file = args.loguru_output_dir + \
+        "/loguru_datalogger_{time}.log"
+
+    logger.add(loguru_output_file,
+               level=args.verbosity, rotation="00:00", compression="zip")
+    logger.warning(
+        "This service is being logged with Loguru with level {args.verbosity} into the file {loguru_output_file}.")
```
Comment (rafaellehmkuhl) [inline services/data_logger/main.py]: Missing f for f-string: `f"This service ..."`

File: services/victron-energy-mppt/main.py [hunk 2]
```diff
@@ -61,15 +59,56 @@ async def root():
...
+
+
+def configure_logging(args: argparse.Namespace):
+    loguru_output_file = args.loguru_output_dir + \
+        "/loguru_datalogger_{time}.log"
+
+    logger.add(loguru_output_file,
+               level=args.verbosity, rotation="00:00", compression="zip")
+    logger.warning(
+        "This service is being logged with Loguru with level {args.verbosity} into the file {loguru_output_file}.")
```
Comment (rafaellehmkuhl) [inline services/victron-energy-mppt/main.py]: Same missing f-string here.

File: services/victron-energy-mppt/main.py [hunk 3]
```diff
@@ -61,15 +59,56 @@ async def root():
         </html>
     """
 
+
 @app.get("/data", response_class=JSONResponse)
 async def data():
     return global_data
 
+
+class LoguruLevelArgumentValidator(argparse.Action):
```
Comment (rafaellehmkuhl) [inline services/victron-energy-mppt/main.py:68]: Maybe this and the `configure_logging` function can be imported from the main datalogger module?

File: services/weatherstation100wx/main.py [hunk 4]
```diff
@@ -57,15 +56,56 @@ async def root():
         </html>
     """
 
+
 @app.get("/data", response_class=JSONResponse)
 async def data():
     return global_data
 
+
+class LoguruLevelArgumentValidator(argparse.Action):
```
Comment (rafaellehmkuhl) [inline services/weatherstation100wx/main.py:65]: Same comments about duplicate entities and f-string.

File: services/sats_comm/main.py [hunk 5]
```diff
@@ -195,11 +197,50 @@ def main():
         except Exception as error:
             logger.exception(error)
         finally:
-            logger.debug(f"Resting for a moment ({REST_TIME} seconds) before next data transmission.")
+            logger.debug(
+                f"Resting for a moment ({REST_TIME} seconds) before next data transmission.")
             time.sleep(REST_TIME)
 
+
+class LoguruLevelArgumentValidator(argparse.Action):
```
Comment (rafaellehmkuhl) [inline services/sats_comm/main.py:315]: Same here.

**Review by rafaellehmkuhl — CHANGES_REQUESTED** [review]

File: services/victron-energy-mppt/main.py (hunk 3 above)
Comment (joaoantoniocardoso) [reply services/victron-energy-mppt/main.py:68]: Yeah, that would be great, but I've tried and I don't know how to deal with imports on Python :eyes:

Comment (rafaellehmkuhl) [reply services/victron-energy-mppt/main.py:68]: Oh, they can be a pain in the ass haha
We are rushing, so I think it's fine to approve that and improve latter.

Comment (joaoantoniocardoso) [reply services/victron-energy-mppt/main.py:68]: yeah, we know the price :laughing:

**Review by rafaellehmkuhl — APPROVED** [review]

**Review by patrickelectric — APPROVED** [review]

**Comment by patrickelectric** [comment]
@joaoantoniocardoso please fix the conflicts

**Comment by joaoantoniocardoso** [comment]
> @joaoantoniocardoso please fix the conflicts

@patrickelectric done

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**